Src/AutoFixtureUnitTest/Kernel/RangedRequestTest.cs

```diff
@@ -0,0 +1,306 @@
+﻿using System;
+using AutoFixture.Kernel;
+using Xunit;
+
+namespace AutoFixtureUnitTest.Kernel
+{
+    public class RangedRequestTest
+    {
+        [Theory]
+        [InlineData(null, typeof(int), 1, 2)]
+        [InlineData(typeof(int), null, 1, 2)]
+        [InlineData(typeof(int), typeof(int), null, 2)]
+        [InlineData(typeof(int), typeof(int), 1, null)]
+        public void ShouldFailWithArgumentExceptionForNullArguments(Type memberType, Type operandType, object min, object max)
+        {
+            // Exercise system and Verify outcome
+            Assert.Throws<ArgumentNullException>(() =>
+                new RangedRequest(memberType, operandType, min, max));
+
+            // Teardown
+        }
+
+        [Fact]
+        public void PropertiesShouldReturnPassedToConstructorValues()
+        {
+            // Fixture setup
+            var memberType = typeof(object);
+            var operandType = typeof(int);
+            var min = 42;
+            var max = 100;
+
+            // Exercise system
+            var sut = new RangedRequest(memberType, operandType, min, max);
+
+            // Verify outcome
+            Assert.Equal(memberType, sut.MemberType);
+            Assert.Equal(operandType, sut.OperandType);
+            Assert.Equal(min, sut.Minimum);
+            Assert.Equal(max, sut.Maximum);
```

Comment:
You could replace the [Assertion Roulette](http://xunitpatterns.com/Assertion%20Roulette.html) here with plain old [Structural Equality](https://en.wikipedia.org/wiki/Relational_operator), like so:

*(Typed on a text-editor)*

```c#
var expected = new RangedRequest(typeof(object), typeof(int), 42, 100);
var actual = new RangedRequest(memberType, operandType, min, max);
Assert.True(expected.Equals(actual));
```
This is also formatted according to [this](http://blog.ploeh.dk/2013/06/24/a-heuristic-for-formatting-code-according-to-the-aaa-pattern/) heuristic, so you don't need to put any comments on each test phase.

Reply:
Thanks for sharing that idea of structural equality. It's definitely a useful practice and I'm already using it in the tests around (driven by samples 😅). Unfortunately, it isn't applicable in this particular test. Here I verify that each value I passed to the constructor equals the corresponding property. The suggested approach wouldn't guarantee that.

As for the test formatting heuristic, I don't agree with that Mark's idea, especially for the OSS projects. In my opinion, in the OSS project all code should be consistent and unified. When somebody creates a PR, he looks at the code around to align the style. If one sees that part of our tests is decorated with comments, while other part is not, he will be confused and will not have a clear understanding of the rules. The arranged comments in a simple test makes much less harm than missing comments in more complex tests. Therefore, if we need to pick the single style, I'd definitely vote to arrange the comments.

Another story is the readability. Personally, I'm used to work with commented tests. When I see a test without comments (we have such ones in the project) I spend much more time to understand it, rather than when comments are present. Therefore, [that idea](http://blog.ploeh.dk/2013/06/24/a-heuristic-for-formatting-code-according-to-the-aaa-pattern/) is a pure personal preference with pros and cons, rather than a strong rule (like tab vs spaces) 😉

P.S. We have a R# template for a test, therefore I don't write those lines manually - they are already present when I create a test.